tuf/api/metadata.py

```diff
@@ -735,6 +799,18 @@ def __init__(
 
         self.roles = roles
 
+    def __eq__(self, other: "Any") -> bool:
+        if not isinstance(other, Root):
+            return False
+
+        return (
+            super().__eq__(other)
+            and self.keys == other.keys
+            and self.roles == other.roles
+            and self.consistent_snapshot == other.consistent_snapshot
+            and self.unrecognized_fields == other.unrecognized_fields
```

Comment:
unrecognized_fields is handled by super() already, correct?

This comment applies to all signed-derivatives